examples/batched-solver/batched-solver.cpp

```diff
@@ -0,0 +1,378 @@
+// with one column/one row. The advantage of this concept is that using
+// multiple vectors is a now a natural extension of adding columns/rows are
+// necessary.
+using value_type = double;
+using real_type = gko::remove_complex<value_type>;
+using index_type = int;
+using size_type = gko::size_type;
+using vec_type = gko::batch::MultiVector<value_type>;
+using real_vec_type = gko::batch::MultiVector<real_type>;
+using mtx_type = gko::batch::matrix::Csr<value_type, index_type>;
+using bicgstab = gko::batch::solver::Bicgstab<value_type>;
+
+
+// @sect3{Convenience functions}
+// Unbatch batch items into distinct Ginkgo types
+namespace detail {
+
+template <typename InputType>
+auto unbatch(const InputType* batch_object)
+{
+    auto unbatched_mats =
+        std::vector<std::unique_ptr<typename InputType::unbatch_type>>{};
+    for (size_type b = 0; b < batch_object->get_num_batch_items(); ++b) {
+        unbatched_mats.emplace_back(
+            batch_object->create_const_view_for_item(b)->clone());
+    }
+    return unbatched_mats;
+}
+
+}  // namespace detail
+
+
+// @sect3{'Application' structures and functions}
+// Structure to simulate application data related to the linear systems
+// to be solved.
+//
+// We use raw pointers below to demonstrate how to handle the situation when
+// the application only gives us raw pointers. Ideally, one should use
+// Ginkgo's gko::Array class here.
+struct ApplSysData {
+    // Number of small systems in the batch.
+    size_type nsystems;
+    // Number of rows in each system.
+    int nrows;
+    // Number of non-zeros in each system matrix.
+    int nnz;
+    // Row pointers for one matrix
+    const index_type* row_ptrs;
+    // Column indices of non-zeros for one matrix
+    const index_type* col_idxs;
+    // Nonzero values for all matrices in the batch, concatenated
+    const value_type* all_values;
+    // RHS vectors for all systems in the batch, concatenated
+    const value_type* all_rhs;
+};
+
+
+// Generates a batch of tridiagonal systems.
+//
+// @param nrows  Number of rows in each system.
+// @param nsystems  Number of systems in the batch.
+// @param exec  The device executor to use for the solver.
+//   Normally, the application may not deal with Ginkgo executors, nor do we
+//   need it to. Here, we use the executor for backend-independent device
+//   memory allocation. The application, for example, might assume Hip
+//   (for AMD GPUs) and use `hipMalloc` directly.
+ApplSysData appl_generate_system(const int nrows, const size_type nsystems,
+                                 std::shared_ptr<gko::Executor> exec);
+
+// Deallocate application data.
+void appl_clean_up(ApplSysData& appl_data, std::shared_ptr<gko::Executor> exec);
+
+
+int main(int argc, char* argv[])
+{
+    // Print the ginkgo version information.
+    if (!(std::string(argv[5]) == "time"))
+        std::cout << gko::version_info::get() << std::endl;
```

Comment:
```suggestion
    if (!(std::string(argv[5]) == "time")) {
        std::cout << gko::version_info::get() << std::endl;
    }
```
the check is a bit weird without explanation